Morphic.Settings.Tests/SettingFinalizerDescriptionTests.cs

```diff
@@ -0,0 +1,94 @@
+// * Canadian Foundation for Innovation
+// * Adobe Foundation
+// * Consumer Electronics Association Foundation
+
+using Morphic.Core;
+using Morphic.Settings.Spi;
+using System.Collections.Generic;
+using System.Text.Json;
+using Xunit;
+
+namespace Morphic.Settings.Tests
+{
+    public class SettingFinalizerDescriptionTests
+    {
+        [Fact]
+        public void TestJsonDeserializeUnknown()
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonElementInferredTypeConverter());
+            options.Converters.Add(new SettingFinalizerDescription.JsonConverter());
+
+            var json = JsonSerializer.Serialize(new Dictionary<string, object?>()
+            {
+                { "type", "invalid" }
+            });
+            var finalizer = JsonSerializer.Deserialize<SettingFinalizerDescription>(json, options);
+            Assert.NotNull(finalizer);
+            Assert.Equal(SettingFinalizerDescription.HandlerKind.Unknown, finalizer.Kind);
+        }
+
+        [Theory]
+        [InlineData("setCursors", SystemParametersInfo.Action.SetCursors, null, null, true)]
+        [InlineData("setCursors", SystemParametersInfo.Action.SetCursors, false, false, true)]
+        [InlineData("setCursors", SystemParametersInfo.Action.SetCursors, false, true, true)]
+        [InlineData("setCursors", SystemParametersInfo.Action.SetCursors, true, false, true)]
+        [InlineData("setCursors", SystemParametersInfo.Action.SetCursors, true, true, true)]
+        [InlineData("shenanigans", SystemParametersInfo.Action.SetCursors, false, false, false)]
+        public void TestJsonDeserializeSPI(string actionString, SystemParametersInfo.Action action, bool? sendChange, bool? updateUserProfile, bool success)
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonElementInferredTypeConverter());
+            options.Converters.Add(new SettingFinalizerDescription.JsonConverter());
+
+            var json = JsonSerializer.Serialize(new Dictionary<string, object?>()
+            {
+                { "type", "com.microsoft.windows.systemParametersInfo" },
+                { "action", actionString },
+                { "send_change", sendChange },
+                { "update_user_profile", updateUserProfile }
+            });
+            var finalizer = JsonSerializer.Deserialize<SettingFinalizerDescription>(json, options);
+            Assert.NotNull(finalizer);
+            if (success)
+            {
+                Assert.Equal(SettingFinalizerDescription.HandlerKind.SystemParametersInfo, finalizer.Kind);
+                Assert.IsType<SystemParametersInfoSettingFinalizerDescription>(finalizer);
+                SystemParametersInfoSettingFinalizerDescription spi = (SystemParametersInfoSettingFinalizerDescription)finalizer;
+                Assert.Equal(action, spi.Action);
+                if (sendChange != null)
```

Comment:
We don't really have a coding style documented, but if we did, it would say "always use curly braces"